pycbc/noise/reproduceable.py

```diff
@@ -103,17 +103,56 @@ def colored_noise(psd, start_time, end_time, seed=0, low_frequency_cutoff=1.0):
     noise : TimeSeries
         A TimeSeries containing gaussian noise colored by the given psd.
     """
-    white_noise = normal(start_time - FILTER_LENGTH, end_time + FILTER_LENGTH,
-                          seed=seed)
-    flen = int(SAMPLE_RATE / psd.delta_f) / 2 + 1
     psd = psd * 1
+    flen = int(SAMPLE_RATE / psd.delta_f) / 2 + 1
+    oldlen = len(psd)
     psd.resize(flen)
-    psd = pycbc.psd.interpolate(psd, 1.0 / white_noise.duration)
-    invpsd = pycbc.psd.inverse_spectrum_truncation(1.0 / psd,
+    # Want to avoid zeroes in PSD.
+    max_val = psd.max()
+    for i in xrange(len(psd)):
+        if i >= (oldlen-1):
+            psd.data[i] = psd[oldlen - 2]
+        if psd[i] == 0:
```

Comment:
This was done to avoid zeroes below the lower-frequency cutoff. The loop may be inefficient, but computing time in this function is dominated by the stuff that follows.